### DISCUSSION TITLE
new discussion view deeplink with pre-selected category

### DISCUSSION THREAD

**Discussion by panva** [community/community#2999] [Discussions] [answered]
I'd like to deep link into the `/discussions/new` view with a query string parameter that would pre-select the category.

**Comment by evi-liu** [comment]
Thanks for the suggestion! Curious to see if others have the same desire so we can gauge impact and prioritization.

**Comment by bosborn** [comment]
As mentioned in https://github.com/github/feedback/discussions/2858#discussioncomment-205109, `?category=` works for categories w/o white space and w/o an Ampersand.

Working:
* https://github.com/bosborn/test/discussions/new?category=general
* https://github.com/bosborn/test/discussions/new?category=ideas

Not Working:
* https://github.com/bosborn/test/discussions/new?category=Q&A
* https://github.com/bosborn/test/discussions/new?category=q&a
* https://github.com/bosborn/test/discussions/new?category=q%26a
* https://github.com/bosborn/test/discussions/new?category=Show%20and%20tell
* https://github.com/bosborn/test/discussions/new?category=showandtell

~~(As per #3013, the category is not persisted for me when the discussion is created... even when pre-selected by url )~~

**Reply by panva** [reply]
@bosborn the category not persisting may be related to #3000

**Reply by BetaHuhn** [reply]
To get it working with categories that contain white space or an ampersand, **use `-`** instead of `&` or `%20`:

So instead of:
- https://github.com/bosborn/test/discussions/new?category=Q&A
- https://github.com/bosborn/test/discussions/new?category=Show%20and%20tell

use:
- https://github.com/bosborn/test/discussions/new?category=Q-A
- https://github.com/bosborn/test/discussions/new?category=Show-and-tell

**Reply by bosborn** [reply]
Nice! Thanks!

**Comment by BetaHuhn** [comment, marked as answer]
You can use `?category=` to pre-select a specific category.

> **Note:** If your category contains white space or an ampersand, use a dash (`-`) instead.

**Examples:**

- https://github.com/github/feedback/discussions/new?category=discussions-feedback
- https://github.com/github/feedback/discussions/new?category=mobile-feedback

**Comment by gr2m** [comment]
related: when I selected one of the categories in the sidebar and then press on the <kbd>New discussion</kbd> button, it would be nice if the same category would be pre-selected in the form.
